### DISCUSSION TITLE
Custom Templates Usage

### DISCUSSION THREAD

**Discussion by rxerium** [projectdiscovery/nuclei#3017] [Q&A] [answered]
Hi there,

I have 2 questions in regards to using a custom template:
1. Is there any way of tracking the progress of a scan? One moment I see the percentage at 65% then 24% then 96%.
2. Is there a formula to calculate the total time the scan will take? I am scanning 55,000 hosts using 143 templates in my workflow file

Thanks

**Comment by ehsandeep** [comment, marked as answer]
using `-stats` is the only option to get idea about the ongoing scan, since there is run time factor involved during scan, i.e unresponsive host getting removed while scanning, this information will be updated accordingly.

**Comment by rxerium** [comment]
I see. On the second part is there a formula to calculate how long a scan will take? If not how long do you estimate it to take given the total hosts/templates? The stats flag doesn't really give me any indications 🤔

**Reply by MetzinAround** [reply]
@hakrishi I think there's too many variables to really give a solid estimate of time: internet speed, connection status, number of templates being used, type of template. Using `-stats` is the closest we can get to at least being able to see what's going on. `-v` MIGHT help, but it also creates screen clutter.

**Reply by MetzinAround** [reply]
Feel free to re-poen this discussion if you. have more questions. 😊

**Reply by rxerium** [reply]
thanks @MetzinAround
